### ISSUE TITLE
`calcite-block`: Make header more accessible

### ISSUE THREAD

**Issue by minhiu** [Esri/calcite-design-system#11707]
**monday.com sync:** #18005818115

### Check existing issues

- [x] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates and reviewed the [Accessibility page](https://developers.arcgis.com/calcite-design-system/foundations/accessibility) for guidance.

### Summary

**Issue**
The `calcite-block` component does not have appropriate name information and the header button is not nested in a heading tag with an appropriate level. Examples include:
Controls that have aria-expanded and a title Collapsed / Expanded that changes according to the state of the control, resulting in a confusing string for the controls' names, also, the controls unique name is assigned through aria-describedby:
- "Layer effects, Adjust blur, highlight, and more" (conveyed as: Collapse button expanded  Layer effects, Adjust blur, highlight, and more)

**User Impact**
Screen reader users will be unable to determine that these controls expand and collapse content. They will also have difficulty determining whether the controls are currently expanded or collapsed.

### Actual Behavior

Described in Summary

### Expected Behavior

- Each block header must be enclosed in a ```<button>``` element.
- Each of these buttons must be enclosed in a heading element of an appropriate level. The ```<button>``` element must be the only child of this heading element.
- When the panel is visible, the header button must have aria-expanded="true". When it is hidden, the header button must have aria-expanded="false".
- The button must have aria-controls set to the ID of the panel container.
- If a panel is visible and locked in the expanded state, its header button must have aria-disabled="true".

### Reproduction Sample

https://developers.arcgis.com/calcite-design-system/components/block-group/

### Reproduction Steps

1. Enable a screen reader.
2. Navigate to the "Layer effects, Adjust blur, highlight, and more" block header button.
3. Note that it is announced as "Expand  button  collapsed   Layer effects, Adjust blur, highlight, and more"
4. Activate the control and note that is conveys: "Collapse, expanded"
5. Open Chrome DevTools.
6. Locate the ```<button>``` and note that it has aria-expanded attribute assigned as well as title="Collapse" which value changes into "Expand" upon activating the control.

### Reproduction Version

3.0

### Priority impact

impact - p2 - want for an upcoming milestone

### Calcite package

- [x] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Business/Community Analyst

**Comment by geospatialem** [comment]
> Blocked by the upcoming third party audit, anticipated for late 2025 - assigning myself to keep track of the effort in late 2025.

The above did not surface during the 2025 audit. @minhiu Do you have additional context on why the above surfaced with the team? It seems from the audit sufficient context was provided on the `block` and `block-group` components when expanded/collapsed, so trying to determine if there could be additional context for the team to consider prior to closing the above out as unplanned.

**Comment by minhiu** [comment]
Hi @geospatialem, I have provided all the information for the `block` component from our audit. Seems like they want the ability to wrap the whole `button` in a header tag:

```
<h2>
 <button aria-controls="content" aria-labelledby="header" aria-expanded="true" class="toggle" id="toggle">
    <header class="header" id="header"><div class="title">
       <span class="heading">Geography</span>
       <div class="description">Current map extent</div>
    </div></header>
    (...)
 </button>
</h2>
```

**Comment by geospatialem** [comment]
> Hi [@geospatialem](https://github.com/geospatialem), I have provided all the information for the `block` component from our audit. Seems like they want the ability to wrap the whole `button` in a header tag:
> 
> ```
> <h2>
>  <button aria-controls="content" aria-labelledby="header" aria-expanded="true" class="toggle" id="toggle">
>     <header class="header" id="header"><div class="title">
>        <span class="heading">Geography</span>
>        <div class="description">Current map extent</div>
>     </div></header>
>     (...)
>  </button>
> </h2>
> ```

Gotcha, thanks for the context, @minhiu! It looks like in exploring with `block`'s `headingLevel` the heading level is not communicated to assistive technologies (where JAWS heading window does not mention the `h2` tags), likely as a result of the Shadow DOM as you mentioned: https://codepen.io/geospatialem/pen/WbxxBYw.

This may be a limitation of ShadowDOM, but will need more investigation from other team members to determine if we can move the `h#` higher in the DOM above the `button`.

However, based on the audit with Level Access this year WRT heading levels this is a low priority issue, so adding the context for this to get picked up in a future milestone for further investigation by Calcite developers to determine next steps - if any.
